Neos.ContentRepository.Core/Classes/SharedModel/Node/NodeIdentity.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neos\ContentRepository\Core\SharedModel\Node;
+
+use Neos\ContentRepository\Core\DimensionSpace\DimensionSpacePoint;
+use Neos\ContentRepository\Core\Factory\ContentRepositoryId;
+use Neos\ContentRepository\Core\SharedModel\Workspace\WorkspaceName;
+
+/**
+ * The content-repository-id content-stream-id dimension-space-point and the node-aggregate-id
+ * Are used in combination to distinctly identify a single node.
+ *
+ * By using the content graph for the content repository
+ * one can build a subgraph with the right perspective to find this node:
+ *
+ *      $workspace = $contentRepository->getWorkspaceFinder()->findOneByName($identity->workspaceName);
+ *      $subgraph = $contentGraph->getSubgraph(
+ *          $workspace->currentContentStreamId,
+ *          $nodeIdentity->dimensionSpacePoint,
+ *          // resolve also all disabled nodes
+ *          VisibilityConstraints::withoutRestrictions()
+ *      );
+ *      $node = $subgraph->findNodeById($nodeIdentity->nodeAggregateId);
```

Comment:
This seems too cumbersome still and IMO folks should not use the `WorkspaceFinder` for things other than finding workspaces (in Neos world people can use the `NodeSerializer` instead, but...).
Maybe we can add some `ContentRepository::findNodeByIdentity(NodeIdentity $nodeIdentity)` or so in a separate PR

Reply:
yeah would prefer another pr as cluttering the `ContentRepository` must probably well discussed :D
Or rather its impossible, as we would no know which content graph implementation to use. There _could_ be multiple ones.